instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/AttributeBindings.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.tracer;
+
+/** Represents the bindings of method parameters to attributes of a traced method. */
+public interface AttributeBindings {
+
+  /**
+   * Indicates that the traced method has no parameters bound to attributes.
+   *
+   * @return {@code true} if the traced method has no bound parameters; otherwise {@code false}
+   */
+  boolean isEmpty();
```

Comment:
It's an optimization to avoid copying the method parameters if none of the parameters are going to be traced as attributes.  This is primarily for the Spring instrumentation where the pointcut syntax isn't sophisticated enough (I believe) to apply based on annotations applied to any of the parameters and where `ProceedingJoinPoint#getArgs` results in allocating a new `Object[]` and copying the actual parameter values into it.